### PR TITLE
Update packages and small improvements

### PR DESCRIPTION
**Pull request by scripthunter7**
- Update some packages
- Remove unused packages
- Improve Jest config
- Use ESLint cache
- Remove unnecessary compatibility table checker script

### CODE REVIEW

**Review by maximtop — APPROVED** [review]

File: jest.config.ts [hunk 1]
```diff
@@ -1,14 +1,10 @@
-import type { Config } from '@jest/types';
+import type { Config } from 'jest';
 
-const config: Config.InitialOptions = {
-    preset: 'ts-jest',
+const config: Config = {
     testEnvironment: 'node',
     testTimeout: 30000,
     testMatch: ['**/test/**/*.test.ts'],
-    // Speed up tests by using SWC instead of Babel
```
Comment (slavaleleka) [inline jest.config.ts:8]: why is the comment removed?

File: package.json [hunk 2]
```diff
@@ -36,72 +36,63 @@
     "scripts": {
         "build": "yarn clean && yarn build-types && yarn rollup --config rollup.config.ts --configPlugin @rollup/plugin-json --configPlugin @rollup/plugin-typescript && yarn clean-types",
         "build-types": "tsc --declaration --emitDeclarationOnly --outdir dist/types",
-        "check-compatibility-tables": "node --no-warnings --loader ts-node/esm --experimental-specifier-resolution=node scripts/check-compatibility-tables.ts",
```
Comment (slavaleleka) [inline package.json:39]: if the script is removed, maybe `.github/workflows/check-compatibility-tables.yml` is not needed anymore

File: jest.config.ts (hunk 1 above)
Comment (scripthunter7) [reply jest.config.ts:8]: Accidentally, I'll fix it

File: package.json (hunk 2 above)
Comment (scripthunter7) [reply package.json:39]: Agree, thanks

File: jest.config.ts (hunk 1 above)
Comment (scripthunter7) [reply jest.config.ts:8]: Fixed in https://github.com/AdguardTeam/AGLint/pull/175/commits/40a380cc1227f168cda55da99f428e93270bdba8

File: package.json (hunk 2 above)
Comment (scripthunter7) [reply package.json:39]: Fixed in https://github.com/AdguardTeam/AGLint/pull/175/commits/c19ed5897caf41f47d1b252c41161a9a46ceee6c

**Review by slavaleleka — APPROVED** [review]

**Review by zhelvis — APPROVED** [review]

**Pull request merged**